src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1080,6 +1080,19 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
             return null;
         }
 
+        internal bool HasMaybeNullWhenOrNotNullWhenAttribute(EntityHandle token, AttributeDescription description, out bool when)
+        {
+            AttributeInfo info = FindTargetAttribute(token, description);
+            if (info.HasValue &&
+                // MaybeNullWhen(bool), NotNullWhen(bool)
```

Comment:
>// MaybeNullWhen(bool), NotNullWhen(bool) [](start = 16, length = 41)

Is the assumption that these are the only attributes this method is going to be called for? Consider adding an assert.